File: projects/spring-boot-restv2/demo/src/main/java/com/sourabh/demo/controller/CancellationController.java
package com.sourabh.demo.controller;

import com.sourabh.demo.service.CancellationService;
import io.swagger.annotations.ApiOperation;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api")
public class CancellationController {
    @Autowired
    CancellationService cancellationService;
    //CancelRoomContro

    @ApiOperation(value = "Enter customer-id to cancel booking.",
            notes = "Provide c.id and based on it booking will be cancelled\n" +
                    "Refund will be calculated.")
    @DeleteMapping("/cancellations/{id}")
    public String cancelRoom(@PathVariable int id){
        return cancellationService.cancelRoom(id);



    }


}

File: projects/spring-boot-rest-kt/hotel-reservation/src/main/kotlin/com/sourabh/hotelreservation/controller/CancellationController.kt
package com.sourabh.hotelreservation.controller

import com.sourabh.hotelreservation.service.CancellationService
import org.springframework.beans.factory.annotation.Autowired
import io.swagger.annotations.ApiOperation
import org.springframework.web.bind.annotation.*


@RestController
@RequestMapping("/api")
class CancellationController {
    @Autowired
    var cancellationService: CancellationService? = null


    @ApiOperation(value = "Enter customer-id to cancel booking.", notes = """Provide c.id and based on it booking will be cancelled
Refund will be calculated.""")
    @DeleteMapping("/cancellations/{id}")
    fun cancelRoom(@PathVariable id: Int): String {
        return cancellationService!!.cancelRoom(id)
    }
}
